code/modules/projectiles/guns/misc/chem_gun.dm

```diff
@@ -44,4 +44,81 @@
 	syringes_left++
 	if(chambered && !chambered.BB)
 		chambered.newshot()
-	last_synth = world.time
\ No newline at end of file
+	last_synth = world.time
+
+/obj/item/gun/chem/advanced
+	name = "advanced syringe gun"
+	desc = "A special syringe gun that can synthesize syringes and reagents to go into the syringe. Fires fifteen unit syringes and will copy any reagents inserted into the internal fifteen unit storage."
+	time_per_syringe = 30
+	syringes_left = 6
+	var/syringetype = "regular"
+	var/syringecapacity = 15
+
+//The parent of this already handles processing on init and destroy
+/obj/item/gun/chem/advanced/Initialize()
+	. = ..()
+	chambered = new /obj/item/ammo_casing/chemgun/advanced(src)
+	create_reagents(60)
+
+/obj/item/gun/chem/advanced/attack_self(mob/user)
+	switch(syringetype)
+		if("regular")
+			time_per_syringe = 50
+			syringecapacity = 20
+			syringetype = "cyro"
+		if("cyro")
+			time_per_syringe = 75
+			syringecapacity = 10
+			syringetype = "piercing"
+		if("piercing")
+			syringecapacity = 60
+			syringetype = "bluespace"
```

Comment:
am i reading this right, if it is regular == cryo syringe, cryo == piercing syringe...